File: linked_list/linked_list_cycle.cpp
class Solution {
public:
    bool hasCycle(ListNode *head) {
        unordered_set<ListNode*> set;
        
        while (head != nullptr) {
            if (set.find(head) != set.end()) {
                return true;
            }
            set.insert(head);
            head = head->next;
        }
        
        return false;
    }
};

File: easy/linked_list/linked_list_cycle.java
/**
 * Definition for singly-linked list. class ListNode { int val; ListNode next;
 * ListNode(int x) { val = x; next = null; } }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        ListNode fast = head;
        ListNode slow = head;

        boolean answer = false;

        while (fast != null && fast.next != null) {
            fast = fast.next.next;
            slow = slow.next;

            if (fast == slow) {
                answer = true;
                break;
            }
        }

        return answer;

    }
}
